From a7983753044bbc823074ad28ea4c3e1b05e3f5e0 Mon Sep 17 00:00:00 2001
From: Abel Vesa <abel.vesa@nxp.com>
Date: Mon, 29 Oct 2018 13:50:18 +0200
Subject: [PATCH 4960/5242] MLK-19966 clk: imx: Skip disabling uart clocks if
 m4 is active

commit  e3f28a4fa3d69de79905833bf6221cd5fb1b9e75 from
https://source.codeaurora.org/external/imx/linux-imx.git

No way of knowing when any of the uart clocks is currently
in use by m4 so just skip the gating for all of them.

Signed-off-by: Abel Vesa <abel.vesa@nxp.com>
Reviewed-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/imx/clk.c b/drivers/clk/imx/clk.c
index 3fc367b..a6103b1 100644
--- a/drivers/clk/imx/clk.c
+++ b/drivers/clk/imx/clk.c
@@ -104,6 +104,9 @@ void __init imx_register_uart_clocks(struct clk ** const clks[])
 
 static int __init imx_clk_disable_uart(void)
 {
+	if (imx_src_is_m4_enabled())
+		return 0;
+
 	if (imx_keep_uart_clocks && imx_uart_clocks) {
 		int i;
 
-- 
1.7.9.5

